### DISCUSSION TITLE
Example fail to run?

### DISCUSSION THREAD

**Discussion by Spammed** [dimforge/nalgebra#1212] [Q&A] [answered]
Perhaps a Rust beginner's mistake.
I do not get the [example](https://docs.rs/nalgebra-sparse/latest/nalgebra_sparse/#example-coo---csr---matrix-vector-product) to work but get:

```
assert_matrix_eq!(csr, dense);
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
|
|     the trait `matrixcompare::Matrix<_>` is not implemented for `CsrMatrix<{float}>`   
|     required by a bound introduced by this call
```
As dependecies i have
```
[dependencies]
approx = "0.5.1"
nalgebra = "0.32.1"

nalgebra-sparse = "0.9"
matrixcompare = "0.3.0"
```

What else do I need to do?

**Comment by Andlon** [comment, marked as answer]
Hi, this is perhaps not so clear from the documentation, but using the `matrixcompare` functionality requires the `compare` feature enabled, so that you have to write

```toml
nalgebra-sparse =  { version="0.9", features = [ "compare" ] }
```

Similarly, you have to do the same with `nalgebra` if you want to compare dense vectors/matrices.

**Comment by Spammed** [comment]
Thank you very much. It works.
